### PR TITLE
Add instructions to add new users to featuretools feedstock 

### PR DESCRIPTION
**Pull request by gsheni**

### CODE REVIEW

File: release.md [hunk 1]
```diff
@@ -99,3 +99,17 @@ In order to release on conda-forge, you can either wait for a bot to create a PR
 1. A bot should automatically create a new PR in [conda-forge/featuretools-feedstock](https://github.com/conda-forge/featuretools-feedstock/pulls) - note, the PR may take up to a few hours to be created
 2. Update requirements changes in `recipe/meta.yaml` (bot should have handled version and source links on its own)
 3. After tests pass, a maintainer will merge the PR in
+
+# Miscellaneous
+## Adding users to featuretools-feedstock
```
Comment (rwedge) [inline release.md:104]:
```suggestion
## Adding maintainers to featuretools-feedstock
```

Comment (gsheni) [reply release.md]: Fixed
https://github.com/alteryx/featuretools/pull/2215/commits/3a1bfb277255c155dec07759c37c5db967d6acf6

**Review by rwedge — APPROVED** [review]
looks good

**Review by bchen1116 — APPROVED** [review]
Nice!

**Pull request merged**